datadog_checks_dev/datadog_checks/dev/tooling/manifest_validator/common/validator.py

```diff
@@ -0,0 +1,273 @@
+            else:
+                self.fail(output)
+
+
+class NameValidator(BaseManifestValidator):
+
+    NAME_PATH = {"v1": '/name', "v2": '/assets/integration/id'}
+
+    def validate(self, check_name, decoded, fix):
+        correct_name = check_name
+
+        path = self.NAME_PATH[self.version]
+        name = decoded.get_path(path)
+
+        if not isinstance(name, str) or name.lower() != correct_name.lower():
+            output = f'  incorrect `name`: {name}'
+            if fix:
+                decoded.set_path(path, correct_name)
+                self.fix(output, f'  new `name`: {correct_name}')
+            else:
+                self.fail(output)
+
+
+class MetricsMetadataValidator(BaseManifestValidator):
+
+    METADATA_PATH = {"v1": "/assets/metrics_metadata", "v2": "/assets/integration/metrics/metadata_path"}
+
+    def validate(self, check_name, decoded, fix):
+        # metrics_metadata
+        path = self.METADATA_PATH[self.version]
+        metadata_in_manifest = decoded.get_path(path)
+
+        metadata_file = get_metadata_file(check_name)
+        metadata_file_exists = os.path.isfile(metadata_file)
+
+        if not metadata_in_manifest and metadata_file_exists:
+            # There is a metadata.csv file but no entry in the manifest.json
+            self.fail('  metadata.csv exists but not defined in the manifest.json of {}'.format(check_name))
+        elif metadata_in_manifest and not metadata_file_exists:
+            # There is an entry in the manifest.json file but the referenced csv file does not exist.
+            self.fail('  metrics_metadata in manifest.json references a non-existing file: {}.'.format(metadata_file))
+
+
+class MetricToCheckValidator(BaseManifestValidator):
+    METRIC_TO_CHECK_EXCLUDE_LIST = {
+        'openstack.controller',  # "Artificial" metric, shouldn't be listed in metadata file.
+        'riakcs.bucket_list_pool.workers',  # RiakCS 2.1 metric, but metadata.csv lists RiakCS 2.0 metrics only.
+    }
+    METADATA_PATH = {"v1": "/assets/metrics_metadata", "v2": "/assets/integration/metrics/metadata_path"}
+    METRIC_PATH = {"v1": "/metric_to_check", "v2": "/assets/integration/metrics/check"}
+    PRICING_PATH = {"v1": "/pricing", "v2": "/pricing"}
+
+    def validate(self, check_name, decoded, _):
+        if not self.should_validate() or check_name == 'snmp' or check_name == 'moogsoft':
+            return
+
+        metadata_path = self.METADATA_PATH[self.version]
+        metadata_in_manifest = decoded.get_path(metadata_path)
+
+        # metric_to_check
+        metric_path = self.METRIC_PATH[self.version]
+        metric_to_check = decoded.get_path(metric_path)
+
+        pricing = decoded.get_path("/pricing") or []
+
+        if metric_to_check:
+            metrics_to_check = metric_to_check if isinstance(metric_to_check, list) else [metric_to_check]
+            if any(p.get('metric') in metrics_to_check for p in pricing):
+                return
```

Comment:
should this be a `continue`?

Reply:
No, since this is part of the validation where we no longer need pricing metrics to be included in the `metadata.csv`.

Reply:
I'm thinking of the case when metric_to_check is a list. We'd want to ignore any pricing metric, but still confirm the others are in metadata.csv. Doesn't this `return` mean if _any_ metric_to_check is a pricing metric, then we skip?